### PR TITLE
feat: [CDS-13677]: Add step wizard flow for adding config files and new connector to add var files form

### PR DESCRIPTION
**Pull request by willd0yle**
##### Summary:
Add step wizard for adding config files as well as allowing the user to create a new connector.
Add new connector flow to add var files step wizard

- Replace old form with multi-part step wizard
- Add new connector wizard sub tasks to add config file form
- Add new connector wizard sub tasks to add val files step wizard form
##### Jira Links:
https://harness.atlassian.net/browse/CDP-20328

##### Screenshots:
Add config wizard
Screenshot 2022-01-28 at 11 00 32
image

Screenshot 2022-01-28 at 11 00 43

Add Var files wizard
image

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by willd0yle** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/TerraformPlan/__tests__/TerraformPlan.test.tsx [hunk 1]
```diff
@@ -820,4 +820,82 @@ describe('Test TerraformPlan', () => {
...
+                  varFile: {
+                    type: 'Inline',
+                    content: 'test'
+                  }
+                }
+              ]
+            }
+          }
+        }}
+        // values?.spec?.configuration?.configFiles?.store?.spec?.folderPath
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/TerraformPlan/__tests__/TerraformPlan.test.tsx:893]: Remove commented code

File: src/modules/75-cd/components/PipelineSteps/TerraformPlan/__tests__/TerraformPlan.test.tsx [hunk 2]
```diff
@@ -820,4 +820,82 @@ describe('Test TerraformPlan', () => {
...
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/TerraformPlan/__tests__/TerraformPlan.test.tsx:853]: Remove commented code

**Review by YounixIO — DISMISSED** [review]

**Comment by willd0yle** [comment]
retrigger typecheck

**Comment by willd0yle** [comment]
retrigger typecheck

**Comment by willd0yle** [comment]
retrigger jest

**Comment by willd0yle** [comment]
trigger cypress

**Comment by willd0yle** [comment]
trigger cypress

**Comment by willd0yle** [comment]
trigger cypress

**Comment by willd0yle** [comment]
trigger cypress

File: src/modules/75-cd/components/PipelineSteps/TerraformPlan/TerraformPlan.tsx [hunk 3]
```diff
@@ -391,41 +539,68 @@ function TerraformPlanWidget(
...
+                      selectedConnector={selectedConnector}
+                      setSelectedConnector={setSelectedConnector}
+                    />
+                    {connectorView ? getNewConnectorSteps() : null}
+                    <TerraformConfigStepTwo
+                      name={getString('cd.configFileDetails')}
+                      isTerraformPlan
+                      isReadonly={readonly}
+                      allowableTypes={allowableTypes}
+                      onSubmitCallBack={(data: any, prevStepData: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/TerraformPlan/TerraformPlan.tsx:572]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
                        onSubmitCallBack={(data: unknown, prevStepData: any) => {
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+    orgIdentifier: string
+    accountId: string
+  }>()
+
+  useEffect(() => {
+    const selectedStore = isTerraformPlan
+      ? data?.spec?.configuration?.configFiles?.store?.type
+      : data?.spec?.configuration?.spec?.configFiles?.store?.type
+    setSelectedConnector(selectedStore)
+  }, [])
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigForm.tsx:84]: Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'data?.spec?.configuration?.configFiles?.store?.type', 'data?.spec?.configuration?.spec?.configFiles?.store?.type', 'isTerraformPlan', and 'setSelectedConnector'. Either include them or remove the dependency array. If 'setSelectedConnector' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
  }, [data?.spec?.configuration?.configFiles?.store?.type, data?.spec?.configuration?.spec?.configFiles?.store?.type, isTerraformPlan, setSelectedConnector])
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarFileList.tsx [hunk 5]
```diff
@@ -42,10 +42,12 @@ interface TfVarFileProps {
   formik: FormikProps<TerraformData>
   isReadonly?: boolean
   allowableTypes: MultiTypeInputType[]
+  getNewConnectorSteps?: any
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarFileList.tsx:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
  getNewConnectorSteps?: unknown
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarFileList.tsx [hunk 6]
```diff
@@ -42,10 +42,12 @@ interface TfVarFileProps {
...
+  setSelectedConnector?: any
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TFVarFileList.tsx:46]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
  setSelectedConnector?: unknown
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx [hunk 7]
```diff
@@ -381,36 +529,61 @@ export default function TerraformEditView(
...
+                            setConnectorView={setConnectorView}
+                            selectedConnector={selectedConnector}
+                            setSelectedConnector={setSelectedConnector}
+                          />
+                          {connectorView ? getNewConnectorSteps() : null}
+                          <TerraformConfigStepTwo
+                            name={getString('cd.configFileDetails')}
+                            isReadonly={readonly}
+                            allowableTypes={allowableTypes}
+                            onSubmitCallBack={(data: any, prevStepData: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx:557]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
                              onSubmitCallBack={(data: unknown, prevStepData: any) => {
```

File: src/modules/75-cd/components/PipelineSteps/TerraformPlan/TfPlanVarFileList.tsx [hunk 8]
```diff
@@ -42,10 +42,12 @@ interface TfVarFileProps {
   formik: FormikProps<TFPlanFormData>
   isReadonly?: boolean
   allowableTypes: MultiTypeInputType[]
+  getNewConnectorSteps?: any
+  setSelectedConnector?: any
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/TerraformPlan/TfPlanVarFileList.tsx:46]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
  setSelectedConnector?: unknown
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx [hunk 9]
```diff
@@ -107,16 +135,129 @@ export default function TerraformEditView(
...
+          connectorInfo={undefined}
+          isStep={true}
+          isLastStep={false}
+          type={connectorType}
+        />
+      </StepWizard>
+    )
   }
+
+  const onStepChange = (arg: StepChangeData<any>): void => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx:248]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
  const onStepChange = (arg: StepChangeData<unknown>): void => {
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigFormHelper.ts [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+  if (type === Connectors.BITBUCKET) {
+    return buildBitbucketPayload
+  }
+  if (type === Connectors.GITLAB) {
+    return buildGitlabPayload
+  }
+  return () => ({})
+}
+
+export const stepTwoValidationSchema = (isTerraformPlan: boolean, getString: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigFormHelper.ts:94]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
export const stepTwoValidationSchema = (isTerraformPlan: boolean, getString: unknown) => {
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigFormHelper.ts [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  buildBitbucketPayload,
+  buildGithubPayload,
+  buildGitlabPayload,
+  buildGitPayload
+} from '@connectors/pages/connectors/utils/ConnectorUtils'
+
+export const AllowedTypes = ['Git', 'Github', 'GitLab', 'Bitbucket']
+export type ConnectorTypes = 'Git' | 'Github' | 'GitLab' | 'Bitbucket'
+
+export const tfVarIcons: any = {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformConfigFormHelper.ts:21]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
export const tfVarIcons: unknown = {
```

**Pull request closed without merging**